src/modules/fancyzones/lib/WindowMoveHandler.cpp

```diff
@@ -298,6 +299,15 @@ void WindowMoveHandlerPrivate::MoveWindowIntoZoneByIndexSet(HWND window, HMONITO
             if (zoneWindow != zoneWindowMap.end())
             {
                 const auto& zoneWindowPtr = zoneWindow->second;
+                // Only process windows located on currently active work area.

```

Comment:
This is the only problematic part with this PR, since it should be possible to modify a custom zoneset that is currently applied to a different VD and it should update the windows position. It is working like that in 0.17.

Reply:
A better fix would be to call `MoveWindowIntoZoneByIndexSet()` after `UpdateActiveZoneSet()` making sure that `MoveWindowIntoZoneByIndexSet()` uses the correct zoneset for each desktop.
But for 0.18 we are good with this fix since, let's open an issue for 0.19.

Reply:
What I meant is:
 - use the same custom layout on VD1
 - position a window in a zone on VD1
 - switch to VD2 and edit the custom layout and apply it to VD2
 - the change also effects VD1 (as expected) since it uses the same custom layout
 - the window on VD1 is not re-positioned

Expected:
 - the window on VD1 should be re-positioned to follow the layout change